service-base/src/main/java/org/eclipse/hono/service/metric/MicrometerBasedMetrics.java

```diff
@@ -395,4 +403,32 @@ private void handleTenantTimeout(final String tenantId) {
 
         vertx.eventBus().publish(Constants.EVENT_BUS_ADDRESS_TENANT_TIMED_OUT, tenantId);
     }
+
+    /**
+     * Tracks the device connection time for the given tenant and records the tracked connection time.
+     * 
+     * @param tenantId The tenant identifier.
+     * @param deviceConnectionsCount The number of device connections.
+     */
+    private void trackDeviceConnectionTime(final String tenantId, final int deviceConnectionsCount) {
+
+        connectionTimeTrackers.compute(tenantId,
+                (tenant, deviceConnectionTimeTracker) -> Optional.ofNullable(deviceConnectionTimeTracker)
+                        .map(tracker -> tracker.updateNoOfDeviceConnections(deviceConnectionsCount))
+                        .orElseGet(() -> {
+                            if (deviceConnectionsCount > 0) {
+                                return DeviceConnectionTimeTracker.Builder
+                                        .createInstance()
+                                        .ForTenant(tenant)
+                                        .withVertx(vertx)
+                                        .withNumberOfDeviceConnections(deviceConnectionsCount)
+                                        .recordUsing(connectionTime -> registry
+                                                .timer(METER_CONNECTIONS_AUTHENTICATED_TIME,
```

Comment:
I wonder if a Timer is the right meter type to use here. Since we are not really timing any particular interaction, I think that simply using a Gauge would work here as well, preventing the higher memory footprint a Timer involves. WDYT?

Reply:
I replaced now with a metric type of Gauge and pushed the changes. Please take a look.